ui/page/main_page.go

```diff
@@ -218,20 +226,48 @@ func (mp *MainPage) OnNavigatedTo() {
 			go mp.WL.MultiWallet.Politeia.Sync()
 		}
 	}
+
+	mp.UpdateBalance()
 }
 
 func (mp *MainPage) setLanguageSetting() {
 	langPre := mp.WL.Wallet.ReadStringConfigValueForKey(load.LanguagePreferenceKey)
 	values.SetUserLanguage(langPre)
 }
 
-func (mp *MainPage) UpdateBalance(fetchExch bool) {
-	if len(mp.totalBalanceUSD) == 0 || fetchExch {
-		load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
-	}
+func (mp *MainPage) updateExchangeSetting() {
 	currencyExchangeValue := mp.WL.Wallet.ReadStringConfigValueForKey(dcrlibwallet.CurrencyConversionConfigKey)
 	mp.usdExchangeSet = currencyExchangeValue == values.USDExchangeValue
+	if mp.usdExchangeSet {
+		go mp.fetchExchangeRate()
+	}
+}
+
+func (mp *MainPage) fetchExchangeRate() {
+	if mp.isFetchingExchangeRate {
+		return
+	}
+	maxAttempts := 7
+	delayBtwAttempts := 2 * time.Second
+	mp.isFetchingExchangeRate = true
+	desc := "for getting dcrUsdtBittrex exchange rate value"
+	attempts, err := components.RetryFunc(maxAttempts, delayBtwAttempts, desc, func() error {
+		return load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
+	})
+	if err != nil {
+		log.Errorf("error fetching usd exchange rate value after %d attempts: %v", attempts, err)
+	} else if mp.dcrUsdtBittrex.LastTradeRate == "" {
+		log.Errorf("no error while fetching usd exchange rate in %d tries, but no rate was fetched", attempts)
+	} else {
+		log.Infof("exchange rate value fetched: %s", mp.dcrUsdtBittrex.LastTradeRate)
+		mp.UpdateBalance()
+		mp.RefreshWindow()
+	}
+	mp.isFetchingExchangeRate = false
+}
 
+func (mp *MainPage) UpdateBalance() {
+	mp.updateExchangeSetting()
```

Comment:
This isn't necessary here. This only needs to be done when the setting has likely changed. My previous comment recommends making this check in `Layout`.